### PR TITLE
Updated handwritten recognition notebook

### PR DESCRIPTION
**Pull request by monshinawatra**
- Error analysis
- Refactor training loops
- Regularization (Dropout)
- Save and load model

### CODE REVIEW

**Review by titipata — APPROVED** [review]
LGTM
